### PR TITLE
Java OIDC client creation for client_secret_basic flow

### PR DESCRIPTION
**Pull request by yovasx2**
This PR introduces the creation of the OidcClient maven package which serves as a Java client for interacting with OpenID Connect providers using the IDPartner class. The gem simplifies the process of authorization, token acquisition, and user information retrieval from OIDC providers by encapsulating the complexities of generating, signing, and verifying JWTs. An example usage of the maven package has been provided in the examples folder for reference.

https://identity-online.atlassian.net/browse/DEV-1312

### CODE REVIEW

File: examples/client-secret-basic/config.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "client_id": "CHANGE_ME_CLIENT_ID",
+  "client_secret": "CHANGE_ME_CLIENT_SECRET",
```
Comment (francisco-idpartner) [inline examples/client-secret-basic/config.json:3]: do we need this? I see you deleted it from the Ruby example

File: examples/client-secret-basic/src/main/java/com/idpartner/AppConfig.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  @JsonProperty("port")
+  private int port;
+
+  public String getClientId() {
+    return clientId;
+  }
+
+  // public void setClientId(String clientId) {
+  //   this.clientId = clientId;
```
Comment (francisco-idpartner) [inline examples/client-secret-basic/src/main/java/com/idpartner/AppConfig.java]: If not needed, should we remove it?

File: examples/client-secret-basic/src/main/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,2 @@
+
+server.port=3001
```
Comment (francisco-idpartner) [inline examples/client-secret-basic/src/main/resources/application.properties]: Can we get this from the config? Or, do we need the config if this is set in this other file?

**Review by francisco-idpartner — APPROVED** [review]
Left a few minor comments, but looks pretty good!

File: examples/client-secret-basic/config.json (hunk 1 above)
Comment (yovasx2) [reply examples/client-secret-basic/config.json:3]: we need it, no i deleted the cookie secret: https://github.com/idpartner-app/ruby_oidc_client/pull/2/files#diff-080e18a09945814359b06497f2bfc633e7ae936c191ee6cbe309fbb66ffeabddL6

Comment (yovasx2) [reply examples/client-secret-basic/config.json:3]: usage: https://github.com/idpartner-app/java-oidc-client/pull/1/files#diff-f42c6871aa4ed1b850c8df634cd489584c0e87d2839f81a933d65ce2a87a1c7bR98

File: examples/client-secret-basic/src/main/resources/application.properties (hunk 3 above)
Comment (yovasx2) [reply examples/client-secret-basic/src/main/resources/application.properties:2]: inverstigating

**Review by francisco-idpartner — APPROVED** [review]

**Pull request merged**